redisinsight/ui/src/slices/app/csrf.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice } from '@reduxjs/toolkit'
+import apiService, { setApiCsrfHeader } from 'uiSrc/services/apiService'
+import { setResourceCsrfHeader } from 'uiSrc/services/resourcesService'
+import { AppDispatch, RootState } from '../store'
+
+const getCsrfEndpoint = () => process.env.RI_CSRF_ENDPOINT || ''
+
+interface CSRFTokenResponse {
+  token: string;
+}
+
+export const initialState: {
+  csrfEndpoint: string;
+  loading: boolean;
+  token: string;
+} = {
+  csrfEndpoint: getCsrfEndpoint(),
+  loading: true,
+  token: '',
+}
+
+const appCsrfSlice = createSlice({
+  name: 'appCsrf',
+  initialState,
+  reducers: {
+    setToken: (state, { payload }: { payload: { token: string } }) => { state.token = payload.token },
+    setLoading: (state, { payload }: { payload: { loading: boolean } }) => { state.loading = payload.loading },
+  }
+})
+
+export const { setLoading, setToken } = appCsrfSlice.actions
+
+export const appCsrfSelector = (state: RootState) => state.app.csrf
+
+export default appCsrfSlice.reducer
+
+export function fetchCsrfToken(
```

Comment:
I suggest rename here to fetchCstfTokenAction and next have 3 dispatches (fetchCsrtToken, fetchCsrtTokenSuccess, fetchCsrtTokenFail)